### PR TITLE
781: handle None from fullmatch reference regex

### LINKED ISSUE #781: Error when converting form with `count-selected()` in `repeat_count`
Error: 'NoneType' object has no attribute 'group'

I've verified that this doesn't happen with pyxform v4.0.0

Forum report with sample form: https://forum.getodk.org/t/error-converting-xlsform-to-xml-using-odk-xlsform-online-count-selected/56741

### PR DESCRIPTION
**Pull request by lindsay-stevens**
Closes #781

#### Why is this the best possible solution? Were any other approaches considered?

- this code branch is for `is_pyxform_reference` but it didn't handle the case where there is no match (looks like reference but has other characters) - in which case the iteration over `outer_matches` would fail and throw an error like "None has no group"
- added test coverage for relevant usages of is_pyxform_reference:
  - repeat_count, now with tests showing common usage pattern outcomes
  - seed param, which had alternate usages already (constant/single ref)

#### What are the regression risks?

Fixes a regression, and tries to avoid others of a similar nature by adding more tests.

#### Does this change require updates to documentation? If so, please file an issue [here](https://github.com/XLSForm/xlsform.github.io) and include the link below.

No

#### Before submitting this PR, please make sure you have:
- [x] included test cases for core behavior and edge cases in `tests`
- [x] run `python -m unittest` and verified all tests pass
- [x] run `ruff format pyxform tests` and `ruff check pyxform tests` to lint code
- [x] verified that any code or assets from external sources are properly credited in comments

### CODE REVIEW

File: tests/test_repeat_count.py [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+                xpq.body_control("q1", "input"),
+                # no instance element for generated *_count item
+                """
+                /h:html/h:head/x:model/x:instance/x:test_name[not(./x:r1_count)]
+                """,
+                # no binding for generated *_count item
+                """
+                /h:html/h:head/x:model[not(./x:bind[@nodeset='/test_name/r1_count'])]
+                """,
+                # repeat references existing count element directly.
```
Comment (lognaturel) [inline tests/test_repeat_count.py:70]: Seems like we can close https://github.com/XLSForm/pyxform/issues/435?

Comment (lindsay-stevens) [reply tests/test_repeat_count.py:70]: yes I think so, I closed it with a [comment here](https://github.com/XLSForm/pyxform/issues/435#issuecomment-3305574903)
